### PR TITLE
Fixed the black blobs of annotations in event plotter. 

### PR DESCRIPTION
**Pull request by Riciardos**
Fix for issue #423.

The timeline of the event is split into sections, and the peaks are placed in the approriate time section.
If a section contains more than or equal to 1 peak, add the biggest peak to the annotation list.
The ones in the annotation list will be annotated, the rest are skipped.
Configurable from config file: set less_black to True if you want this fix.
Number of sections also changable from config: lower number should lead to less black blobs(as more peaks are skipped per section)

### CODE REVIEW

File: pax/plugins/io/Plotting.py [hunk 1]
```diff
@@ -179,6 +206,11 @@ def plot_waveform(self, event,
                                       connectionstyle="angle3,"
                                                       "angleA=0,"
                                                       "angleB=-90")
+                if less_black:
+                    #Go to next peak if not the biggest in section, otherwise annotate!
```
Comment (tunnell) [inline pax/plugins/io/Plotting.py]: __[flake8]__
* [E265] block comment should start with '# '

**Comment by JelleAalbers** [comment]
Hi Ricardo,

Thanks again for looking into a pax issue! This is an important improvement for the event display.

Sometimes it is quite useful to annotate peaks even when they are close together: e.g. when an S1 and an S2 are close together, or when we have two large S2s near to each other. Maybe you could complement your fix with an option (on by default) to always annotate the largest X peaks, or the largest X peaks of each peak type, whether or not they are the first in their 

Make you can show an example event display after your fix? You can drag & drop images into the text editing field when you're writing a comment.

(if you fix the lint errors, the Travis builds might still fail; some dependency conflict apparently emerged recently. I hope it gets resolved upstream, else we'll have to eg. pin/downgrade matplotlib as Chris suggested here: https://github.com/XENON1T/pax/pull/426#issuecomment-248409738)

**Comment by Riciardos** [comment]
Hey Jelle,

I have changed some stuff around and I think you will like it this way. I re-factored the annotation_filter into a separate function which reads out a list of options from the config file. 

The main S1 and S2 are always annotated, if they are present.
`'largest' ` will do what it did before: only annotate the largest peak in a section
`'largest_s1s'` or `'largest_s2s'` : will do the same but only for s1s or s2s respectively
`'all_s1s'` or `'all_s2s'` : will annotate all s1s or all s2s (duh)

and any of these can be combined, so for example if you want all the s1s annotated but only the largest s2s, you can set `annotation_filter = ['all_s1s','largest_s2s']` and it will do exactly that. 
Just setting it `'largest'` will give the best readability, and like I said earlier, the main S1 and S2 are always annotated.

I don't have access to Xenon1T data so I had to use the example file in pax, and here is the first event without any filter:
even1nofilter

and here it is with the 'largest' option enabled

event1withfilter

Also for the Travis stuff, I did change the .yml file, but I think it might have something to do with the fact I imported `attrgetter` from the operator module, which is not a built-in function,but the module should be included in every Python environment. Anyway, it's checking it now and we'll see what happens.

**Comment by tunnell** [comment]
@JelleAalbers you find with this?

**Comment by JelleAalbers** [comment]
It would be good to add an option to never hide the largest (say) 5 s1s and s2s in the event, otherwise we may not spot a double scatter.

But I guess I won't have time to look at that for the moment, and this definitely improves the event display. Right now the peak annotations are almost useless in high-energy events. So we can merge this and add the option to not hide the largest x S1s / S2s another time.

**Comment by tunnell** [comment]
Approved. Good, let's merge.  Can always improve method later.

Warning: if two peaks too close, hides label.

**Comment by sanderbreur** [comment]
Ricardo thanks for making this possible.

S

Op di 25 okt. 2016 om 12:53 schreef Christopher Tunnell <
notifications@github.com>:

**Pull request merged**